### PR TITLE
Fix to allow debugger thread setting counter values on FunctionBody

### PR DESCRIPTION
**Pull request by leirocks**

### CODE REVIEW

**Comment by leirocks** [comment]
@curtisman @agarwal-sandeep can you review please?

File: lib/Runtime/Base/CompactCounters.h [hunk 1]
```diff
@@ -207,7 +206,7 @@ namespace Js
         template<typename FieldT>
         void AllocCounters(T* host)
         {
-            Assert(ThreadContext::GetContextForCurrentThread() || ThreadContext::GetCriticalSection()->IsLocked());
+            Assert(ThreadContext::GetContextForCurrentThread() || ThreadContext::GetCriticalSection()->IsLocked() || host->IsInDebugMode());
```
Comment (agarwal-sandeep) [inline lib/Runtime/Base/CompactCounters.h]: Can also happen during detach, so this will be false

**Comment by curtisman** [comment]
LGTM

**Comment by agarwal-sandeep** [comment]
Looks good to me

**Comment by leirocks** [comment]
@dotnet-bot test this please

**Pull request merged**